src/samplers/multinomial.jl

```diff
@@ -52,7 +52,7 @@ function _rand!(rng::AbstractRNG, s::MultinomialSampler,
                 x::AbstractVector{<:Real})
     n = s.n
     k = length(s)
-    if n^2 > k
+    if n > sqrt(k)
```

Comment:
It's just a heuristic but to ensure that we get the same behaviour as before I think we need
```suggestion
    if n > isqrt(k)
```
here (see, e.g., the discussion here: https://github.com/JuliaLang/julia/issues/4884).